src/rgw/rgw_rest_s3.cc

```diff
@@ -3525,26 +3557,47 @@ void RGWRestoreObj_ObjStore_S3::send_response()
 
   rgw::sal::Attrs attrs = s->object->get_attrs();
   auto attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
+  auto expire_iter = attrs.find(RGW_ATTR_RESTORE_EXPIRY_DATE);
```

Comment:
Probably its good to read & save these attrs in RGWRestoreObj::execute() before calling handle_cloudtier_obj(). That way we can identify if its the first restore request as attrs will not be set by then.

Reply:
Makes sense! will try out that.